src/Tizen.NUI/src/public/WindowSystem/QuickPanelClient.cs

```diff
@@ -0,0 +1,441 @@
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        ~QuickPanelClient()
+        {
+            if (!isDisposeQueued)
+            {
+                isDisposeQueued = true;
+                DisposeQueue.Instance.Add(this);
+            }
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <exception cref="MemberAccessException">Thrown when private memeber is a corrupted.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void Dispose()
+        {
+            if (isDisposeQueued)
+            {
+                Dispose(DisposeTypes.Implicit);
+            }
+            else
+            {
+                Dispose(DisposeTypes.Explicit);
+                GC.SuppressFinalize(this);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(DisposeTypes disposing)
+        {
+            int res;
+            if (!disposed)
+            {
+                if (_tzshQpClient != IntPtr.Zero)
+                {
+                    res = Interop.QuickPanelClient.Destroy(_tzshQpClient);
+                    try
+                    {
+                        _tzsh.ErrorCodeThrow(res);
+                    }
+                    catch (ArgumentException e)
+                    {
+                        throw new MemberAccessException("QuickPanelClient is a corrupted");
+                    }
+                    _tzshQpClient = IntPtr.Zero;
+                }
+                disposed = true;
+            }
+        }
+
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        private void OnVisibleChanged(int type, IntPtr ev_info, IntPtr data)
+        {
+            int res;
+            int state;
+
+            res = Interop.QuickPanelClient.GetEventVisible(ev_info, out state);
+            _tzsh.ErrorCodeThrow(res);
+
+            _visibleChanged?.Invoke(this, (StateVisible)state);
+        }
+
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        private void OnOrientationChanged(int type, IntPtr ev_info, IntPtr data)
+        {
+            int res;
+            int state;
+
+            res = Interop.QuickPanelClient.GetEventOrientation(ev_info, out state);
+            _tzsh.ErrorCodeThrow(res);
+
+            _orientationChanged?.Invoke(this, (StateOrientation)state);
+        }
+
+        /// <summary>
+        /// Gets the visible state of the quickpanel.
+        /// </summary>
+        /// <returns>The visible state of the quickpanel service window.</returns>
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation or no service.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public StateVisible GetVisible()
+        {
```

Comment:
How about making this "get only property"? it would look better, I think...

Reply:
I changed this to property named Visible